product_supplierinfo_for_customer/models/product_product.py

```diff
@@ -85,8 +76,8 @@ def price_compute(self, price_type, uom=False, currency=False, company=False):
             partner_id = self.env.context.get(
                 "partner_id", False
             ) or self.env.context.get("partner", False)
-            if partner_id and isinstance(partner_id, models.BaseModel):
-                partner_id = partner_id.id
+            if partner_id and isinstance(partner_id, int):
+                partner_id = self.env["res.partner"].browse(partner_id)
```

Comment:
Why changing that?

Reply:
`_select_customerinfo` like standard `_select_seller` requires a res.partner as argument (later in the funcion it does partner.parent_id), therefore we need to pass the record, not the id

Reply:
Then don't call the variable `partner_id`, but `partner`.